corehq/messaging/scheduling/views.py

```diff
@@ -1085,3 +1092,48 @@ def post(self, request, *args, **kwargs):
             msg[0](request, msg[1])
 
         return self.get(request, *args, **kwargs)
+
+
+@login_and_domain_required
+def messaging_image_upload_view(request, domain):
+    if request.method == 'POST' and request.FILES.get('upload'):
+        image_file = request.FILES['upload']
+
+        if image_file.size > MAX_IMAGE_UPLOAD_SIZE:
+            return JsonResponse({
+                'error': {
+                    "message": _('Image file is too large. Images must be smaller than 1MB'),
+                }
+            }, status=400)
+
+        if image_file.content_type not in VALID_EMAIL_IMAGE_MIMETYPES:
+            image_extensions = [mimetype.split("/")[1] for mimetype in VALID_EMAIL_IMAGE_MIMETYPES]
+            return JsonResponse({
+                'error': {
+                    "message": _('You can only upload {image_extensions} images').format(
+                        image_extensions=", ".join(image_extensions))
+                },
+            }, status=400)
+
+        image = EmailImage.save_blob(
+            image_file,
+            domain=request.domain,
+            filename=image_file.name,
+            content_type=image_file.content_type,
+            delete_after=datetime.utcnow() + timedelta(days=365),
+        )
+        return JsonResponse({
+            'url': absolute_reverse("download_messaging_image", args=[domain, image.blob_id])
+        }, status=201)
+    return JsonResponse({'error': {"message": _('Invalid request')}}, status=400)
+
+
+def messaging_image_download_view(request, domain, image_key):
```

Comment:
This is deliberately globally accessible, right?  Could you make a note of that in a comment?

Reply:
Correct. I'll add a note.

Reply:
https://github.com/dimagi/commcare-hq/pull/33798/commits/19165c39fc06697c1c9cc85ff3a64a2e43b49a1e